tools/cksum.c

```diff
@@ -0,0 +1,129 @@
+/* crc32 -- calculate and print POSIX checksums for a specific string.
+
+   This tool is a stipped clone of cksum tool from Linux coreutils.
+   Is using the same crc32 lookup table crctab as coreutils.
+   Is necessary for ifupdown-ng when the cksum tools is not avaiable on
+   a system and we need VRRP name generation.
+
+   Copyright (C) 2023 EasyNetDev.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
```

Comment:
I don't believe it would be possible to merge the code from coreutils because the GPL-3 code is not compatible with the license of ifupdown-ng itself.

Reply:
Hi,

And what is the solution for this? I took parts of the cksum CRC32 tool and adapted here because on embedded devices where you need to strip a lot of thinks from the OS there is a possibility for not having cksum tool. CRC32 implementation in `sh` is almost impossible.

Reply:
I've changed the license for all files to this:
```
/*
 * tools/crc32.c
 * Purpose: calculate and print POSIX checksums for a specific string.
 *
 * This tool is a clone of cksum tool from Linux coreutils.
 * Is using the same crc32 lookup table crctab as coreutils.
 * Is necessary for ifupdown-ng when the cksum tools is not avaiable on
 * a system and we need VRRP name generation.
 *
 * Copyright (c) 2023 EasynetDev in the Public Benefit, Inc.
 *
 * Permission to use, copy, modify, and/or distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * This software is provided 'as is' and without any warranty, express or
 * implied.  In no event shall the authors be liable for any damages arising
 * from the use of this software.
 */

#define _GNU_SOURCE
```

I will change the the sources to get a string up to 64 bytes as an argument. In this way I will change the sources enough to be a clone of cksum. Is ok like this?